docs/okeps/okep-5193-user-defined-networks.md

```diff
@@ -0,0 +1,1468 @@
+additional networks, these flows will be modified to steer traffic to the correct GR-&lt;network>’s patch port.
+
+When a host process or host networked pod on a Kubernetes node initiates a connection to a service, iptables rules will
+DNAT the nodeport or loadbalancer IP into the cluster IP, and then send the traffic via br-ex where it is masqueraded
+and sent into the OVN GR. These flows can all be modified to detect the service IP and then send to the correct
+GR-&lt;network> patch port. For example, in the br-ex (breth0) bridge today we have flows that match on packets sent
+to the service CIDR (10.96.0.0/24):
+
+```
+[root@ovn-worker ~]# ovs-ofctl dump-flows breth0 table=0 | grep 10.96
+ cookie=0xdeff105, duration=22226.373s, table=0, n_packets=41, n_bytes=4598, idle_age=19399,priority=500,ip,in_port=LOCAL,nw_dst=10.96.0.0/16 actions=ct(commit,table=2,zone=64001,nat(src=169.254.169.2))
+```
+
+Packets that are destined to the service CIDR are SNAT'ed to the masquerade IP of the host (169.254.169.2) and then
+sent to the dispatch table 2:
+
+```
+[root@ovn-worker ~]# ovs-ofctl dump-flows breth0 table=2
+ cookie=0xdeff105, duration=22266.310s, table=2, n_packets=41, n_bytes=4598, actions=mod_dl_dst:02:42:ac:12:00:03,output:"patch-breth0_ov"
+```
+
+In the above flow, all packets have the dest MAC address changed to be that of the OVN GR, and then sent on the patch port
+towards the OVN GR. With multiple networks, host access to cluster IP service flows will now be modified to be on a per
+cluster IP basis. For example, if we assume two services exist on two user defined namespaces with cluster IPs 10.96.0.5
+and 10.96.0.6. The flows would look like:
+
+```
+[root@ovn-worker ~]# ovs-ofctl dump-flows breth0 table=0 | grep 10.96
+ cookie=0xdeff105, duration=22226.373s, table=0, n_packets=41, n_bytes=4598, idle_age=19399,priority=500,ip,in_port=LOCAL,nw_dst=10.96.0.5 actions=set_field:2->reg1,ct(commit,table=2,zone=64001,nat(src=169.254.169.2))
+ cookie=0xdeff105, duration=22226.373s, table=0, n_packets=41, n_bytes=4598, idle_age=19399,priority=500,ip,in_port=LOCAL,nw_dst=10.96.0.6 actions=set_field:3->reg1,ct(commit,table=2,zone=64001,nat(src=169.254.169.2))
+```
+
+The above flows are now per cluster IP and will send the packet to the dispatch table while also setting unique register
+values to differentiate which OVN network these packets should be delivered to:
+
+```
+[root@ovn-worker ~]# ovs-ofctl dump-flows breth0 table=2
+ cookie=0xdeff105, duration=22266.310s, table=2, n_packets=41, n_bytes=4598, reg1=0x2 actions=mod_dl_dst:02:42:ac:12:00:05,output:"patch-breth0-net1"
+ cookie=0xdeff105, duration=22266.310s, table=2, n_packets=41, n_bytes=4598, reg1=0x3 actions=mod_dl_dst:02:42:ac:12:00:06,output:"patch-breth0-net2"
+```
+
+Furthermore, host networked pod access to services will be restricted to the network it belongs to. For more information
```

Comment:
we don't support host-net pods or host pieces today... but I guess you want to keep these bits as is?

Reply:
yeah lets keep it as is so we dont lose the original intention of the design, but i'll put it as future goals